client/blocks/inline-help/popover.jsx

```diff
@@ -159,9 +169,14 @@ class InlineHelpPopover extends Component {
 						),
 						[ VIEW_RICH_RESULT ]: (
 							<InlineHelpRichResult
-								result={ this.state.selectedResult }
 								setDialogState={ setDialogState }
 								closePopover={ onClose }
+								type={ selectedResult?.[ RESULT_TYPE ] ?? RESULT_ARTICLE }
+								title={ selectedResult?.[ RESULT_TITLE ] }
+								link={ selectedResult?.[ RESULT_LINK ] }
+								description={ selectedResult?.[ RESULT_DESCRIPTION ] }
+								tour={ selectedResult?.[ RESULT_TOUR ] }
+								postId={ selectedResult?.post_id }
```

Comment:
Oh, I get it now 🙂 Why don't you just pass a `result` object and do the field accesses inside the component?

By the way, defining constants for field _names_ is very unusual, I wouldn't mind removing them. And keep constants only for field _values_.

Reply:
Should be fixed in https://github.com/Automattic/wp-calypso/pull/56063/commits/be9954e756a27e1da1178dedf85fd66aaf1efd65 – these constants were also used in [`my-sites/customer-home/cards/features/help-search/index.jsx`](https://github.com/Automattic/wp-calypso/blob/trunk/client/my-sites/customer-home/cards/features/help-search/index.jsx#L6-L9), I updated that file accordingly (cc @p-jackson). In https://github.com/Automattic/wp-calypso/pull/56063/commits/a9926bce0720c10083be5184b13908204d848fb7 I removed those unused constants entirely.